src/shared/Microsoft.Git.CredentialManager.Tests/HostProviderRegistryTests.cs

```diff
@@ -57,5 +83,140 @@ public void HostProviderRegistry_MultipleValidProviders_ReturnsFirstRegistered()
 
             Assert.Same(provider1Mock.Object, result);
         }
+
+        [Fact]
+        public void HostProviderRegistry_GetProvider_ProviderSpecified_ReturnsProvider()
+        {
+            var context = new TestCommandContext
+            {
+                Settings = {ProviderOverride = "provider3"}
+            };
+            var registry = new HostProviderRegistry(context);
+            var input = new InputArguments(new Dictionary<string, string>());
+
+            var provider1Mock = new Mock<IHostProvider>();
+            var provider2Mock = new Mock<IHostProvider>();
+            var provider3Mock = new Mock<IHostProvider>();
+            provider1Mock.Setup(x => x.Id).Returns("provider1");
+            provider1Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(true);
+            provider2Mock.Setup(x => x.Id).Returns("provider2");
+            provider2Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(true);
+            provider3Mock.Setup(x => x.Id).Returns("provider3");
+            provider3Mock.Setup(x => x.IsSupported(It.IsAny<InputArguments>())).Returns(false);
+
+            registry.Register(provider1Mock.Object, provider2Mock.Object, provider3Mock.Object);
+
+            IHostProvider result = registry.GetProvider(input);
```

Comment:
Probably missing something basic, 3 providers are registered only the last returns false.  Why would we expect to get the last back?   Is it because we set provider3 in Settings, hence the false doesn't matter?

Reply:
We get the last provider back because it's provider3 and the test is checking that if we set the override to provider3 that it will be returned, even if by the 'auto' detection logic provider3 is NOT valid.